bangc-ops/mlu_op.h

```diff
@@ -11474,6 +11474,84 @@ mluOpRoiAlignBackward_v2(mluOpHandle_t handle,
                          const mluOpTensorDescriptor_t grads_image_desc,
                          void *grads_image);
 
+// Group:MsDeformAttnForward
+/*!
+ * @brief Implements a multi-scale deformable attention module used in Deformable-Detr.
+ * For detailed information about Deformable-Detr, see "Deformable DETR: Deformable
+ * Transformers for End-to-End Object Detection."
+ *
+ * @param[in] handle
+ * Handle to an MLUOP context that is used to manage MLU devices and queues
+ * in ::mluOpMsDeformAttnForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] data_value_desc
+ * The descriptor of data_value. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] data_value
+ * Pointer to the MLU memory that stores the input multi-scale feature maps.
+ * @param[in] data_spatial_shapes_desc
+ * The descriptor of data_spatial_shapes. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] data_spatial_shapes
+ * Pointer to the MLU memory that stores the shapes of multi-scale feature maps.
+ * @param[in] data_level_start_index_desc
+ * The descriptor of data_level_start_index. For detailed information,
+ * see ::mluOpTensorDescriptor_t.
+ * @param[in] data_level_start_index
+ * Pointer to the MLU memory that stores the feature maps offset in data_value.
+ * @param[in] data_sampling_loc_desc
+ * The descriptor of data_sampling_loc. For detailed information,
+ * see ::mluOpTensorDescriptor_t.
+ * @param[in] data_sampling_loc
+ * Pointer to the MLU memory that stores the normalized coordinates of sample points.
+ * @param[in] data_attn_weight_desc
+ * The descriptor of data_attn_weight. For detailed information,
+ * see ::mluOpTensorDescriptor_t.
+ * @param[in] data_attn_weight
+ * Pointer to the MLU memory that stores the attention weight.
+ * @param[in] im2col_step
+ * The value of im2col_step.
+ * @param[in] data_col_desc
+ * The descriptor of the output tensor. For detailed information,
```

Comment:
The descriptor of the tensor \b data_col .

Reply:
done